### PR TITLE
add configurationTarget when updating settings

### PR DESCRIPTION
**Pull request by AndrewDant**
Currently all settings updates must be done at the workspace level, this adds the option to target the global level while remaining backwards compatible

### CODE REVIEW

**Comment by AndreasArvidsson** [comment]
Don't forget to update the args example/description: https://github.com/AndreasArvidsson/andreas-vscode/blob/da5b873a2c85dee27b2efea83a8c17dbae757934/src/commands/commands.ts#L156

File: src/commands/settings.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import { workspace } from "vscode";
+import { workspace, ConfigurationTarget } from "vscode";
 
 export function getSetting<T>(section: string, defaultValue?: T): T | undefined {
     return workspace.getConfiguration().get(section, defaultValue);
 }
 
-export function setSetting(section: string, value: any) {
-    return workspace.getConfiguration().update(section, value);
+export function setSetting(section: string, value: any, configurationTarget?: boolean | ConfigurationTarget) {
```
[the remark this answers is not in the document]
Comment (AndrewDant) [reply src/commands/settings.ts:7]: I might be missing some context regarding typescript enums. I can pass true/false and this works but if I pass the strings for the enum properties (eg 'Global') it doesn't work.

[the remark this answers is not in the document]
Comment (AndreasArvidsson) [reply src/commands/settings.ts:7]: You need to do the conversion in a switch statement. I would do something like this:
```ts
import * as vscode from "vscode";

export function getSetting<T>(section: string, defaultValue?: T): T | undefined {
    return vscode.workspace.getConfiguration().get(section, defaultValue);
}

type ConfigurationTarget = "global" | "workspace" | "workspaceFolder";

export function setSetting(section: string, value: any, configurationTarget?: ConfigurationTarget) {
    return vscode.workspace
        .getConfiguration()
        .update(
            section,
            value,
            configurationTarget != null ? getConfigurationTarget(configurationTarget) : undefined
        );
}

function getConfigurationTarget(
    configurationTarget: ConfigurationTarget
): vscode.ConfigurationTarget {
    switch (configurationTarget) {
        case "global":
            return vscode.ConfigurationTarget.Global;
        case "workspace":
            return vscode.ConfigurationTarget.Workspace;
        case "workspaceFolder":
            return vscode.ConfigurationTarget.WorkspaceFolder;
    }
}

```

[the remark this answers is not in the document]
Comment (AndrewDant) [reply src/commands/settings.ts:7]: This implementation is case sensitive but I don't get any kind of useful output if the configurationTarget isn't a valid value. Is there in easy update we can make for better logging?

[the remark this answers is not in the document]
Comment (AndrewDant) [reply src/commands/settings.ts:7]: Talked to pokey a bit and I see that this is sort of normal/expected behavior for an extension like this. I could write it to accept any type and log an error if it isn't something valid, but I'll wait to hear if you think I should do something like that or operate with the assumption that the right type will be passed in.

[the remark this answers is not in the document]
Comment (AndreasArvidsson) [reply src/commands/settings.ts:7]: Yeah you can add something like this:
```ts
default:
            // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
            throw new Error(`Unknown configuration target: ${configurationTarget}`);
```

[the remark this answers is not in the document]
Comment (AndrewDant) [reply src/commands/settings.ts:7]: Perfect, that's simpler than where I was going with it.

**Pull request merged**